app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -16,42 +28,59 @@ import { CaveatTypes } from '../../../../../shared/constants/permissions';
 import { UNKNOWN_TICKER_SYMBOL } from '../../../../../shared/constants/app';
 import { PermissionNames } from '../../../controllers/permissions';
 import { getValidUrl } from '../../util';
+import {
+  EndApprovalFlow,
+  GetCaveat,
+  GetChainPermissionsFeatureFlag,
+  RequestPermittedChainsPermission,
+  RequestUserApproval,
+  SetActiveNetwork,
+} from './types';
 
-export function findExistingNetwork(chainId, findNetworkConfigurationBy) {
+export function findExistingNetwork(
+  chainId: Hex,
+  findNetworkConfigurationBy: (options: { chainId: Hex }) => void,
+) {
   if (
     Object.values(BUILT_IN_INFURA_NETWORKS)
-      .map(({ chainId: id }) => id)
+      .map(({ chainId: id }) => id as Hex)
       .includes(chainId)
   ) {
     return {
       chainId,
       ticker: CURRENCY_SYMBOLS.ETH,
-      nickname: NETWORK_TO_NAME_MAP[chainId],
-      rpcUrl: CHAIN_ID_TO_RPC_URL_MAP[chainId],
-      type: CHAIN_ID_TO_TYPE_MAP[chainId],
+      nickname:
+        NETWORK_TO_NAME_MAP[chainId as keyof typeof NETWORK_TO_NAME_MAP],
+      rpcUrl:
+        CHAIN_ID_TO_RPC_URL_MAP[
+          chainId as keyof typeof CHAIN_ID_TO_RPC_URL_MAP
+        ],
+      type: CHAIN_ID_TO_TYPE_MAP[chainId as keyof typeof CHAIN_ID_TO_TYPE_MAP],
     };
   }
   return findNetworkConfigurationBy({ chainId });
 }
 
-export function validateChainId(chainId) {
+export function validateChainId(chainId: Hex) {
   const _chainId = typeof chainId === 'string' && chainId.toLowerCase();
```

Comment:
`_chainId` shouldn't contain a boolean type. All of the other validate functions return the validated data, not the result in boolean.

```suggestion
  const _chainId = chainId.toLowerCase();
```

`Hex` validates at the type level that `chainId` is a subtype of `string`. If that's not enough, we should throw an error at runtime.

```suggestion
  if (typeof chainId !== 'string') {
    throw ethErrors.rpc.invalidParams({
      message: `Expected 0x-prefixed, unpadded, non-zero hexadecimal string 'chainId'. Received:\n${chainId}`,
    });
  }
  const _chainId = chainId.toLowerCase();
```